emails: Add quote prefix ">" in plain text missed message emails.

sahil839 opened the pull request:
We add quote prefix ">" to each line of the message in the plain text
missed message emails, which are then rendered as quotes by email
clients. We also move the message content in the next line after sender.

This helps us in clearly showing the message authors in missed message
emails especially in emails with multiple messages and senders.

Fixes #15836.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing Plan:** <!-- How have you tested? --> Tested by configuring SMTP settings and using Thunderbird mail client for plain-text version.


**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
For stream messages-
![Screenshot (170)](https://user-images.githubusercontent.com/35494118/89342728-d2632800-d6c0-11ea-8082-5c5f18eac8c3.png)
For PMs-
![Screenshot (172)](https://user-images.githubusercontent.com/35494118/89342975-32f26500-d6c1-11ea-9320-0e71936db9ea.png)
For Group PMs-
![Screenshot (173)](https://user-images.githubusercontent.com/35494118/89343090-646b3080-d6c1-11ea-8382-74175e2d4576.png)

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

sahil839 commented:
I have just kept the quotes for PMs also even though we do not show sender in PMs just to maintain consistency with other emails.
We can change it to just show messages without quoting if we want to do so.

timabbott commented on zerver/lib/email_notifications.py:
```diff
@@ -166,6 +166,14 @@ def fix_spoilers_in_text(content: str, language: str) -> str:
             output.append(line)
     return '\n'.join(output)
 
+def add_quote_prefix_in_text(content: str) -> str:
```
Can you add a comment here?

timabbott commented:
This looks reasonable to me implementation-wise; I'd like to see feedback from @rhn on the format before we merge.

Gittenburg commented:
I don't think the quoting makes sense for PMs.

sahil839 commented:
@Gittenburg I am also not sure of quoting in PMs. I have included it just for maintaining consistency with other emails. We can change it according to feedback from others.

sahil839 commented:
Added the comment as asked above. 
Also added a commit to rename the tests. The previous names were of the form `*_personal_missed_stream_messages`, which I feel was not correct and thus changed it to `*_missed_personal_messages` and have done similar changes for huddle too.

andersk commented:
I’m worried that this quoting may interact badly with email clients that strip out or collapse quoted sections in summary or threaded views.

timabbott commented:
Are there email clients that do that when they hadn't received similar content previously?  Gmail at least hides duplicated content in those, which wouldn't be a problem here.

rhn commented:
This looks actually better than I expected, and seems totally functional.

I'm not sure about collapsing, I think it may be worth checking what GMail does. Perhaps collapsing clients default to HTML anyway.

timabbott commented:
Thinking about this more, I think it would probably be a bug in the email client if it hid a region prefixed with `>` that hadn't appeared elsewhere in the thread -- because that's a common situation you get when someone replies to an original message you weren't copied on CCing you.

timabbott commented:
That said, if we're worried about this issue, we could generate a div with our own CSS for this sort of thing.

Gittenburg commented:
If we generate HTML for this the proper tag to use is `<blockquote>`, which is more accessible than a div since it's semantic.

sahil839 commented:
Rebased the PR.

We are quoting the content in plain-text version. So, I guess we cannot use `div` or `blockquote` here.

timabbott commented:
Merged, thanks @sahil839!
